Automattic/woocommerce-payments
# RPP: Extract gateway methods

Fixes #7496

#### Changes proposed in this Pull Request

While working on a minimum checkout solution (https://github.com/Automattic/woocommerce-payments/pull/7332) for the new payment process, it turned out that intent **metadata** and **level 3 data** are generated within in the gateway class, but independent of it, and required for the payment process. This PR focuses on moving:

1. Metadata generation to the new `OrderService`: It's a simple 32-line method.
2. Level 3 data generation to a new `Level3Service`, as it's much more elaborate.

After extracting those from the gateway, their usages were replaced with calls to the services through `wcpay_get_container()`. This is not ideal, as they should be proper dependencies, but once the payment process refactoring is done, those calls would be deleted.

#### Testing instructions

Perform test checkout on a US store, and verify in the Stripe dashboard that all data is correct.

Tests have been changed as well:

- There are new ones (100% coverage, as always) for the services.
- Existing gateway tests do not focus on level 3 and meta data anymore, instead only ensuring that the right request setter has been called. 

This is the reason the PR has so many changes.

-------------------
